pandas/io/json/_normalize.py

```diff
@@ -234,7 +235,10 @@ def _pull_field(js, spec):
             for field in spec:
                 result = result[field]
         else:
-            result = result[spec]
+            try:
+                result = result[spec]
```

Comment:
huh? again I really really don't like try/excepts they hide all kinds of errors. library code is NOT like user code where try/excepting is the norm. we *must* be very explicit.

Reply:
I gotcha entirely, but I was struggling how to accomplish this here. `is_integer` doesn't cut it here as `"1"` is valid for the list bit is not an integer according `is_integer`. If you have a suggestion that gets rid of this I'll gladly drop this.

Reply:
what case are you trying to get here? e.g. what is result and what is the spec?

Reply:
@jreback I think I got it. Thanks for pushing for the right fix btw.